pkg/functions/function_test.go

```diff
@@ -398,3 +401,129 @@ func TestFunction_Stamp(t *testing.T) {
 		t.Fatal("expected journal log not found")
 	}
 }
+
+func TestFunction_NewFunctionLocalSettings(t *testing.T) {
+	root, rm := Mktemp(t)
+	defer rm()
+	fConfig := fn.Function{Root: root, Runtime: "go", Name: "f"}
+	client := fn.New(fn.WithBuilder(mock.NewBuilder()), fn.WithRegistry(TestRegistry))
+	f, err := client.Init(fConfig)
+	if err != nil {
+		t.Fatal(err)
+	}
+	f.Local.Remote = true
+
+	err = f.Write()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Load the function from the same location
+	f, err = fn.NewFunction(root)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if !f.Local.Remote {
+		t.Fatal("expected remote flag to be set")
+	}
+}
+
+func TestFunction_LocalSettingsGit(t *testing.T) {
```

Comment:
I think that if you take the time to create a good documentation string (as the very first thing done), then it helps with deciding on the name of the test, and also helps keep focused.  Maybe try giving these two tests a good comment first, and see if that helps with choosing a good name!